samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemFactory.scala

```diff
@@ -106,8 +106,8 @@ class KafkaSystemFactory extends SystemFactory with Logging {
     if (appConfig.getAppMode == ApplicationMode.BATCH) {
       val streamConfig = new StreamConfig(config)
       streamConfig.getStreamIds().filter(streamConfig.getIsIntermediateStream(_)).map(streamId => {
+        // only the override here
         val properties = new Properties()
-        properties.putAll(streamConfig.getStreamProperties(streamId))
```

Comment:
It's not obvious to me how this is different from the kafkaSpec.getProperties(). Will follow up with you

Reply:
I believe `kafkaSpec.getProperties()` strictly returns kafka properties. Did we want to avoid a subset of the stream properties?

Reply:
ok. make sense now